tcblib/slack_functions.py

```diff
@@ -0,0 +1,52 @@
+import boto3
+import hmac
+import slack
+
+def validate_request(req):
+    """Validate slack request to prevent the hax."""
+    slack_signing_secret = boto3.client('ssm').get_parameter(
+        Name='/slack/tcb/signing_secret',
+        WithDecryption=True
+    )['Parameter']['Value']
+    signing_secret = str.encode(slack_signing_secret)
+
+    slack_request_timestamp = req['headers']['X-Slack-Request-Timestamp']
+    request_body = req['body']
+
+    token = 'v0:{0}:{1}'.format(
+        slack_request_timestamp,
+        request_body
+    )
+    encoded_token = str.encode(token)
+
+    slack_signature = req['headers']['X-Slack-Signature']
+
+    encrypted_secret = hmac.new(
+        key=signing_secret,
+        msg=encoded_token,
+        digestmod='sha256'
+    ).hexdigest()
+
+    expected_signature = 'v0={0}'.format(encrypted_secret)
+
+    if not hmac.compare_digest(expected_signature, slack_signature):
+        return False
```

Comment:
Pls error handle better tyyy

Reply:
Honestly I think its best to just return a 401 with no message at all.